### ISSUE TITLE
.vsto/C# solution freeze when calling Microsoft.SharePoint.Client.ClientContext cc = authManager.GetContext(sharePointSiteUrl)

### ISSUE THREAD

**Issue by GitHubUserJB** [pnp/pnpframework#961]
Our code is calling
**var authManager = new PnP.Framework.AuthenticationManager(appClient, certificate, tenantID);
using(Microsoft.SharePoint.Client.ClientContext cc = authManager.GetContext(sharePointSiteUrl)){}**

Up until recently, it worked fine.
SharePoint Online documents was retrieved to our application.
However, a few weeks ago the call has worked sporadically.
Sometimes the call went through. Other times the code just froze without entering catch-exception.

Today we are faced with a new issue.
SharePoint Online documents are not retrieved whatsoever. The call fails every time.
The following code just fails without any catch-exception error message.
**Microsoft.SharePoint.Client.ClientContext cc = authManager.GetContext(sharePointSiteUrl)**

Our application has "site.selected", and the specific SharePoint site has read permission for the application.
We have also given the application "SharePoint" -> "Sites.Read.All" to prevent missing privilegues from causing the issue.

The issue seems to occur without any changes to code or app permissions.

We have tried using "GetContextAsync()" as described in https://github.com/pnp/pnpframework/issues/447
However, this did not resolve the issue. Same error occured with code freezing and no catch-exception error message.

Is there a way to solve this, so we can retrieve SharePoint Online documents in our application?

**Comment by radiocold** [comment]
Hi Mister, do you have any solution? I have same problem.

**Comment by kco-jyskebank-dk** [comment]
Yes. The following code line solved our issue. Hope it solve yours :D
using (ClientContext cc = new PnP.Framework.AuthenticationManager().GetACSAppOnlyContext(sharePointSiteUrl, appClient, clientSecret))
